File: Python/220530_2.py
"""1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
"""
        
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        
        def dfs(node):
            if not node:
                return None
            elif node.val == target.val:
                return node
            return dfs(node.left) or dfs(node.right)
        
        return dfs(cloned)

File: JS/220530_2.js
//1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} original
 * @param {TreeNode} cloned
 * @param {TreeNode} target
 * @return {TreeNode}
 */

var getTargetCopy = function (original, cloned, target) {
  dfs = function (node) {
    if (node === null) {
      return null;
    } else if (node.val == target.val) {
      return node;
    }
    return dfs(node.left) || dfs(node.right);
  };

  return dfs(cloned);
};
